### PR TITLE
Add phase for machine creating

### LINKED ISSUE #398: Delete a cluster failed to complete
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

 ./clusterctl delete cluster -p examples/openstack/out/provider-components.yaml  --kubeconfig kubeconfig --bootstrap-type kind
I0716 01:04:36.306411   14851 clusterdeployer.go:160] Creating bootstrap cluster
I0716 01:04:36.306480   14851 createbootstrapcluster.go:27] Preparing bootstrap cluster
I0716 01:05:25.546045   14851 clusterdeployer.go:168] Pivoting Cluster API stack to bootstrap cluster
I0716 01:05:25.546086   14851 pivot.go:67] Applying Cluster API Provider Components to Target Cluster
I0716 01:05:26.841692   14851 pivot.go:72] Pivoting Cluster API objects from bootstrap to target cluster.

**What did you expect to happen:**
delete should end with machines and all rescources successfully deleted

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by jichenjc**
when doing creating machine, we can should those phase info.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #398 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/openstack/machine/actuator.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ const (
 	RetryIntervalInstanceStatus = 10 * time.Second
 
 	TokenTTL = 60 * time.Minute
+
+	MachineCreating string = "Creating"
+	MachineRunning  string = "Running"
+	MachineFailed   string = "Failed"
```
Comment (hidekazuna) [inline pkg/cloud/openstack/machine/actuator.go]: Is machine phase name common? Creating/Running/Faild are not in vm_state. If we follow OpenStack vm_state, The value would be Building/Active/Error.
https://developer.openstack.org/api-ref/compute/?expanded=create-server-detail,list-servers-detail

**Comment by jichenjc** [comment]
I will take another look at this, long story short, delete cluster action will wait for machines to have status, but no provider didn't set that, so we have to add this PR and something like following to set status as workaround..

```
        if err != nil {
                ic.updatePhase(ctx, machine, MachineFailed)
                return fmt.Errorf("Guest %s encode status failed in cluster %s", machine.Name, cluster.Name)
        }
        machine.Status.ProviderStatus = ext
```

**Comment by sbueringer** [comment]
I'm not sure if we should still pursue this. The phases of the machine are managed completely by the general machine-controller of ClusterAPI with v1alpha2

**Comment by jichenjc** [comment]
yes

**Pull request closed without merging**